packages/framework/src/Support/DataCollection.php

```diff
@@ -45,49 +48,75 @@ class DataCollections extends Collection
      *
      * Each Markdown file will be parsed into a MarkdownDocument with front matter.
      *
-     * @return DataCollections<string, \Hyde\Markdown\Models\MarkdownDocument>
+     * @return static<string, \Hyde\Markdown\Models\MarkdownDocument>
      */
     public static function markdown(string $name): static
     {
-        static::needsDirectory(static::$sourceDirectory);
-
-        return new static(static::findFiles($name, 'md')->mapWithKeys(function (string $file): array {
-            return [static::makeIdentifier($file) => MarkdownFileParser::parse($file)];
-        }));
+        return static::discover($name, 'md', function (string $file): MarkdownDocument {
+            return MarkdownFileParser::parse($file);
+        });
     }
 
     /**
      * Get a collection of YAML documents in the resources/collections/<$key> directory.
      *
      * Each YAML file will be parsed into a FrontMatter object.
      *
-     * @return DataCollections<string, \Hyde\Markdown\Models\FrontMatter>
+     * @return static<string, \Hyde\Markdown\Models\FrontMatter>
      */
     public static function yaml(string $name): static
     {
-        static::needsDirectory(static::$sourceDirectory);
+        return static::discover($name, ['yaml', 'yml'], function (string $file): FrontMatter {
+            $content = Filesystem::getContents($file);
 
-        return new static(static::findFiles($name, ['yaml', 'yml'])->mapWithKeys(function (string $file): array {
-            return [static::makeIdentifier($file) => MarkdownFileParser::parse($file)->matter()];
-        }));
+            $content = Str::between($content, '---', '---');
+
+            try {
+                $parsed = Yaml::parse($content) ?: [];
+            } catch (ParseException $exception) {
+                throw new InvalidArgumentException(sprintf("Invalid YAML in file: '%s' (%s)", $file, $exception->getMessage()), previous: $exception);
```

Comment:
Was wondering if we need to add the message from the parse error here, and I think we do, because the previous exception does show up in the realtime compiler, but not when running the build command

![image](https://github.com/hydephp/develop/assets/95144705/a908d612-4231-471f-ad5e-638042d8304a)




![image](https://github.com/hydephp/develop/assets/95144705/725b52ad-5365-41e7-a327-e354a111858e)